Adds a kitchen baster

TobleroneSwordfish opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[FEATURE] [BALANCE]
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Adds a kitchen baster as suggested here: https://forum.ss13.co/showthread.php?tid=18175
It functions like a syringe with 100 units capacity that injects 50 units at a time, BUT can only be used on food. It can also be click-drag filled.

![image](https://user-images.githubusercontent.com/20713227/156373810-1daa0ef8-afa3-4fa6-9e47-e0759b91edb1.png)
Currently one can be bought from the kitchen vendor both for stealing considerations and because I am scared of mapping.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Syringes are clunky and time consuming to use for good reason, but making it easier for food specifically seems okay.
Numbers could easily change, I am not a chef expert.

## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete it.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```changelog
(u)LeahTheTech
(*)Added a kitchen baster for adding pois- DELICIOUS ingredients to food.
```

Kyle2143 commented:
At a glance the code looks pretty good and it's addition in the chef vending machine is a good way to add it to the round. 
Was hoping it would be comically though, the sprite looks to be the same size as a regular 15u syringe except it has no point... Though we could always change it later for clarity so it doesn't look like the chef is walking around with a medical syringe, so it's not a big deal for me.

TobleroneSwordfish commented:
Yeah the sprite size is me being lazy and not wanting to have to duplicate all the syringe reagent colour masks, I can fix that. For the in-hand sprite I quite like the idea that you can't tell at a glance whether the chef is holding a syringe or not, could make for some interesting bluffs.

NightmareChamillian commented:
I feel like 25 u injections would be better, since committing to 50 entire units in one click is a little bit much, both for the chef and people eating the foods.

TobleroneSwordfish commented:
Okay I changed it to 25 units and because I am physically incapable of leaving well alone, I also refactored it to work with 4 icon stages so it didn't look weird.

frawhst commented on code/modules/chemistry/Chemistry-Tools.dm:
```diff
@@ -101,7 +101,7 @@ ABSTRACT_TYPE(/obj/item/reagent_containers)
 				return
 		// First filter out everything we don't want to refill or empty quickly.
-		if (!istype(over_object, /obj/item/reagent_containers/glass) && !istype(over_object, /obj/item/reagent_containers/food/drinks) && !istype(over_object, /obj/reagent_dispensers) && !istype(over_object, /obj/item/spraybottle) && !istype(over_object, /obj/machinery/plantpot) && !istype(over_object, /obj/mopbucket) && !istype(over_object, /obj/item/reagent_containers/mender) && !istype(over_object,  ...
+		if (!istype(over_object, /obj/item/reagent_containers/glass) && !istype(over_object, /obj/item/reagent_containers/food/drinks) && !istype(over_object, /obj/reagent_dispensers) && !istype(over_object, /obj/item/spraybottle) && !istype(over_object, /obj/machinery/plantpot) && !istype(over_object, /obj/mopbucket) && !istype(over_object, /obj/item/reagent_containers/mender) && !istype(over_object,  ...
```
Can this be a static list or something? This line makes me eyes hurt. Or even just add some line breaks (`\`)

TobleroneSwordfish replied:
I've made it a static list, let me know if the way I'm checking it is dumb

frawhst commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -286,10 +295,29 @@
+			if (S_DRAW)
+				if (!istype(target, /obj/item/reagent_containers))
+					return
+			if (S_INJECT)
```
Would like some kind of user feedback here- something like "You can't fit the nozzle in that!" so people don't get confused when nothing is happening

frawhst reviewed (approved):
Syringe cleanup stuff is much appreciated. Seems good, might want to add more to the vendor or give this to sous chefs on spawn or something, but that can be for later

TobleroneSwordfish commented:
!merge_upstream

frawhst commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -286,10 +295,31 @@
+					return
+			if (S_INJECT)
+				if (!istype(target, /obj/item/reagent_containers/food) && !istype(target, /obj/item/reagent_containers/glass))
+					boutput(user, "<span class='alert'>You can't fit [src]'s nozzle in that</span>")
```
```suggestion
					boutput(user, "<span class='alert'>You can't fit [src]'s nozzle in that.</span>")
```

frawhst commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -286,10 +295,31 @@
+		switch (mode)
+			if (S_DRAW)
+				if (!istype(target, /obj/item/reagent_containers))
+					boutput(user, "<span class='alert'>You can't fit [src]'s nozzle in that</span>")
```
```suggestion
					boutput(user, "<span class='alert'>You can't fit [src]'s nozzle in that.</span>")
```

frawhst reviewed (commented):
i clearly don't check my requested reviews enough
